src/frontend/src/session.rs

```diff
@@ -579,6 +582,8 @@ pub struct SessionImpl {
 
     /// Last idle instant
     last_idle_instant: Arc<Mutex<Option<Instant>>>,
+
+    cursor_manager: Arc<tokio::sync::Mutex<CursorManager>>,
```

Comment:
IIUC, there cannot be concurrent queries running in the same session. Therefore, there is no need to use a aysnc mutex here. We can just use a normal mutext like what we did in `txn`.